Code/GraphMol/RGroupDecomposition/RGroupUtils.cpp

```diff
@@ -103,5 +105,53 @@ bool hasDummy(const RWMol &core) {
   }
   return false;
 }
+
+namespace {
+std::string MolToText(const ROMol &mol) {
+  bool hasQuery = false;
+  for (const auto atom : mol.atoms()) {
+    if (atom->hasQuery() && atom->getQuery()->getDescription() != "AtomNull") {
+      hasQuery = true;
+      break;
+    }
+  }
+  if (!hasQuery) {
+    for (const auto bond : mol.bonds()) {
+      if (bond->hasQuery()) {
+        hasQuery = true;
+        break;
+      }
+    }
+  }
+  if (!hasQuery) {
+    return MolToSmiles(mol);
+  } else {
+    return MolToSmarts(mol);
+  }
+}
 }  // namespace
 
+std::string toJSON(const RGroupRow &rgr, const std::string &prefix) {
```

Comment:
We should probably add RGroupCols as well I think.

Reply:
I had been thinking about this primarily as a tool for human-readable tests, not unit testing, and I think the row-based format is way better for that.
But if we're looking at it as generic functionality for converting this stuff to JSON (which it clearly is) then having the column version completely makes sense.
On it.